wp-cli/i18n-command
# Add support for indirect translation function calls in JS

This change allows makepot on JS files to work for babel transpiled files. This change is similar to https://github.com/wp-cli/i18n-command/pull/204 but related to a babel behavior instead.

In [some cases](https://github.com/babel/babel/blob/91fb64eaf53c67a7ed7605aab71f8dcaa4d39ae2/packages/babel/src/transformation/modules/lib/remaps.js#L32), babel will convert simple function calls to indirect ones (in the form of `(0, __)( 'translate' )`).

I guess this is done so that legacy ES5 script that uses strict mode are able retrieve the global `window` object using `this`.

In any case, babel often transpiles code like:
```js
import { __ } from '@wordpress/i18n';

__( 'translate' );
```

Into:

```js
var _i18n = _$$_REQUIRE(_dependencyMap[4], "@wordpress/i18n");

(0, _i18n.__)( 'translate' );
```

### Testing instructions
- Download this JS file which uses `@wordpress/i18n` and is transpiled by babel (bundled by metro.js): https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/bundle/android/App.js
- Run `wp i18n make-pot . --include=App.js -ignore-domain gutenberg-android.pot`
- Notice strings are correctly extracted with this patch, otherwise the POT file is empty
